suzieq/db/parquet/parquetdb.py

```diff
@@ -389,64 +389,48 @@ def migrate(self, table_name: str, schema: SchemaForTable) -> None:
-                        try:
-                            namespace = item.split('namespace=')[1] \
-                                .split('/')[0]
-                        except IndexError:
-                            # Don't convert data not in our template
-                            continue
-
-                        df = pd.read_parquet(item)
-                        df['sqvers'] = sqvers
-                        df['namespace'] = namespace
-                        newdf = migrate_rtn(df)
-
-                        cols = newdf.columns
-                        # Ensure all fields are present
-                        for field in schema_def:
-                            if field not in cols:
-                                newdf[field] = defvals.get(schema_def[field],
-                                                           '')
-
-                        newdf.drop(columns=['namespace', 'sqvers'])
-
-                        newitem = item.replace(f'sqvers={sqvers}',
-                                               f'sqvers={current_vers}')
-                        newdir = os.path.dirname(newitem)
-                        if not os.path.exists(newdir):
-                            os.makedirs(newdir, exist_ok=True)
-
-                        table = pa.Table.from_pandas(
-                            newdf, schema=schema.get_arrow_schema(),
-                            preserve_index=False)
-                        pq.write_to_dataset(table, newitem,
-                                            version=PARQUET_VERSION,
-                                            compression="ZSTD",
-                                            row_group_size=100000)
-                        self.logger.debug(
-                            f'Migrated {item} version {sqvers}->'
-                            f'{current_vers}')
-                        os.remove(item)
-
-                    rmtree(
-                        f'{self._get_table_directory(table_name, True)}/'
-                        f'sqvers={sqvers}', ignore_errors=True)
+                dataset = self._get_cp_dataset(table_name, True, sqvers,
+                                               'all', '', '')
+                if not dataset:
+                    continue
+
+                for file in dataset.files:
+                    try:
+                        df = ds.dataset(file,
+                                        format='parquet',
+                                        partitioning='hive') \
+                            .to_table() \
+                            .to_pandas(self_destruct=True)
+                    except pa.ArrowInvalid as e:
+                        self.logger.warning(f'Unable to migrate file: {e}')
+
+                    if migrate_rtn:
+                        df = migrate_rtn(df, table_name, schema)
+                    # Always perform generic migration
+                    df = generic_migration(df, table_name, schema)
```

Comment:
Can you explain why always generic migration?

Reply:
This is to make things more robust. The generic migration allows us to always make sure that all the fields have the right type and if it is not the case, we try the conversion. I think there could be two different approaches:
- Let the specific migration function handle everything, so we run the generic_migration only when we don't have a migration function to run
- Simple conversions might not be handled by the specific migration functions, so we always the the generic one

Reply:
I think the correct approach would be to keep the code as is, otherwise we need to take care of every possible problems in the specific migration function